src/core/net/ip6.cpp

```diff
@@ -963,7 +975,7 @@ Error Ip6::HandlePayload(Message &          aMessage,
     {
 #if OPENTHREAD_CONFIG_TCP_ENABLE
     case kProtoTcp:
-        error = mTcp.ProcessReceivedSegment(*message, aMessageInfo);
+        error = mTcp.ProcessReceivedSegment(aIp6Header, *message, aMessageInfo);
```

Comment:
```suggestion
        error = mTcp.HandleMessage(aIp6Header, *message, aMessageInfo);
```

Keep the naming consistent with UDP and ICMP6?